app/views/events/_snapshot_event_row.html.erb

```diff
@@ -11,7 +11,7 @@
       <%= link_to storage_locs.find { |i| i.id == loc.id}.name, storage_location_path(loc.id) %>
     <td>
       <% loc.items.values.each do |entry| %>
-        <%= link_to items.find { |i| i.id == entry.item_id}.name, item_path(entry.item_id) %>:
+        <%= link_to items.find { |i| i.id == entry.item_id}&.name || "Item #{entry.item_id} (deleted)", item_path(entry.item_id) %>:
```

Comment:
Don't you also have to apply this same fix in app/views/events/_event_row.html.erb as well?

Reply:
Also does it make sense to create a link to a deleted item? Won't it just lead to a 404?

Reply:
And looking into the future... maybe we would want to implement soft deletes for items, so that we could grab the items name here. Because it doesn't seem like displaying "Item 1234 (deleted)" will be very helpful for users.

Reply:
@coalest, we only allow deletes per se on items that have had no transactions associated with them.  Otherwise we allow deactivates if the inventory level on all storage locations for the item is 0.

This is dealing with a particular edge case .   If we deleted an item that existed on the snapshot, it was making the event run crash.    
It reall is an edge case, but could possibly happen again,  as the intent is to do periodic refersher snapshots.